src/data/boards.js

```diff
@@ -0,0 +1,190 @@
+      throw new Error("Board description must not be empty.");
+    fields.description = newBoard.description;
+  }
+
+  return fields;
+}
+
+
+module.exports = {
+  /**
+   * Get a board with a specific ID.
+   * @param {String} id The board's id.
+   * @returns A board.
+   */
+  readByID: async (id) => {
+    if(!error_handler.checkObjectId(id))
+      throw new Error("id is not valid.");
+
+    const boardCollection = await boards();
+    const board = await boardCollection.findOne({_id: ObjectId(id)});
+    if(board === null)
+      throw new Error("There is no board with that id.");
+
+    board['_id'] = board['_id'].toString();
+
+    return board;
+  },
+
+  /**
+   * Gets all boards that a user is a part of.
+   * @param {String} id The user's id.
+   * @returns A list of board objects.
+   */
+  readAll: async (id) => {
+    if(!error_handler.checkObjectId(id))
+      throw new Error("id is not valid.");
+
+    const userCollection = await users();
+    const user = await userCollection.findOne({_id: ObjectId(id)});
+    if(user === null)
+      throw new Error("There is no user with that id.");
+    const boardList = user['boards'];
+
+    return boardList;
+  },
+
+  /**
+   * Adds a new board to the collection. Picks a random color from pre-approved values, and starts the description, lists, and cards as empty.
+   * @param {String} userId The ID of the user requesting to create a new board.
+   * @param {String} boardName The name of the new board.
+   * @returns The board object.
+   */
+  create: async (userId, boardName) => {
+    if(!error_handler.checkObjectId(userId))
+      throw new Error("id is not valid.");
+
+    if(!error_handler.checkNonEmptyString(boardName))
+      throw new Error("Board name must not be empty.");
+
+    const boardColor = colors[Math.floor(Math.random() * colors.length)];
+    const description = "";
+    const members = [userId];
+    const lists = [];
+    const cards = [];
+
+    let newBoard = {
+      boardName,
+      boardColor,
+      description,
+      members,
+      lists,
+      cards
+    };
+
+    const boardCollection = await boards();
+    const insertInfo = await boardCollection.insertOne(newBoard);
+    if(insertInfo.insertedCount === 0)
+      throw new Error("Could not add board.");
+
+    return await module.exports.readByID(insertInfo.insertedId.toString());
+  },
+
+  /**
+   * Updates the board with the fields of boardData from a PUT request.
+   * @param {String} id The board's id.
+   * @param {Object} boardData An object that may contain boardName, boardColor, and description.
+   * @returns A board object with updated fields.
+   */
+  update: async (id, boardData) => {
+    if(!error_handler.checkObjectId(id))
+      throw new Error("id is not valid.");
+
+    const boardCollection = await boards();
+    let updatedBoard = newFields(boardData, await module.exports.readByID(id));
+    if(Object.keys(updatedBoard).length === 0)
+      console.log("No fields to update.");
```

Comment:
Is this how we are handling? maybe should be returning here as to not progress to the rest of the code.